doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,345 @@
+RIOT community. RIOT is the Friendly IoT Operating System: it is friendly to
+users, allowing them to easily build and deploy IoT solutions that are stable
+and trouble-free.
+
+Below are the design philosophies that are typically followed by developers to
+achieve this, including descriptions of tradeoffs between them and where the
+resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't collecting data.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler. They shouldn't demand that users do
+the same.  They should, however, allow users to manage power through different
+modes and functions.
+
+#### Small memory footprint
+
+Most of RIOT's targeted use cases are well addressed by devices in class 1 of
+the taxonomy presented in [1]. If small price differences are important or the
+energy budget is particularly tight, the available memory might be near the
+bottom of this class. Over-the-air updating currently reduces the available ROM
+by over half.
+
+RIOT should provide out-of-the-box support for devices with ~10 KiB of
+available RAM and ~100 KiB of ROM. It should be just as possible to address
+real use cases involving even smaller devices, starting at ~2 KiB of RAM and
+~10 KiB of ROM, with manual optimizations and specialised modules. Users should
+be given the choice of what they want to spend their memory on.
+
+#### Constrained networking
+
+RIOT should deliver best-in-class communication bandwidth and robustness.
+
+Network stacks should remain up-to-date as relevant standards emerge; they
+should be adequately extensible to support this. Users should be able to adapt
+them and choose between different stacks; they should be able to get the best
+performance and most relevant functionality out of whatever resources they have
+available.
+
+## Short learning curve
+
+RIOT's use cases involve makers, researchers in non-computer science fields,
+broadly skilled engineers making proofs-of-concept in IoT startups, and
+experienced embedded C developers.
+
+All of these users should be comfortable using RIOT and its tooling, whatever
+their platform or hardware. No user of any skill level should be tripped up
+because they haven't configured something that they wouldn't reasonably be
+expected to know about. Whatever the user's background, RIOT should demand as
+little RIOT-specific learning as possible. It should, therefore, adhere to
+common systems and networking standards such as C99, POSIX, libc, UDP, CoAP,
+IPv6, and so on wherever it makes sense.
```

Comment:
This list appears a bit too arbitrary - maybe cut it?
"It should, therefore, adhere to common systems and networking standards"